earthspy/earthspy.py

```diff
@@ -188,6 +194,19 @@ def set_query_parameters(
 
         return None
 
+    def get_raster_compression(self, compress_mode) -> str:
+        """Verify valid keyword for raster compression
+        
+        :return: Compression mode
+        """
+        
+        if compress_mode in ['DEFLATE','LZW','PACKBITS','JPEG', 'WEBP','LZMA','ZSTD']:
+           self.compress_mode = compress_mode
+        else:
+          raise KeyError("Compression mode not found")
+        
+        return self.compress_mode
```

Comment:
There's an issue here where `self.compress_mode` is only set if `compress_mode` matches one of the strings in the list. If it's set to `None`, which it is by default, the method returns `self.compress_mode` but it actually doesn't exist since it didn't enter the condition. We need an extra condition in an `elif` before `else` where `self.compress_mode` is set to `None` if compress_mode is `None`. Then `self.compress_mode` exists in all cases.

Reply:
Also, it would be great if you could rename `compress_mode` and `self.compress_mode` to `raster_compression` and `self.raster_compression` to be consistent with the input in `set_query_parameters`